Some bug fixes to the new posterior plots

cdcapano opened the pull request:
We found a couple bugs while using the new `plot_posterior` and `plot_movie` added in PR #1050: the marginal plots do not always have the same limits as the scatter/density plots, there is a large white space between the last plot and the colorbar when no marginal plots are created. This patch fixes these two issues and introduces a few new things:
 * When just creating a 2D plot, the marginal plots are made to be 1/3 the density/scatter plot and the right one is rotated. I've found this is useful when creating plots for presentations and/or papers. (See 2D density plot example, below.)
 * The colorbar is automatically turned on/off depending on whether a `z-arg` is specified, rather than making the user do it.
 * Minimum and maximum values to plot for each parameter can be specified on the command line. This is useful for comparing posteriors from different runs.
 * If all values for a parameter have a common offset > 1000, that offset is removed automatically and added to the parameter label. This is so you don't have to specify `tc-11...` on the command line whenever you want to plot coalescence time. See the `tc` axes in the example plots, below.
 * Output files made by `plot_movie` now have the sample number in their name rather than frame number, and the sample number is appropriately zero-padded to make for easy sorting in bash when feeding to `ffmpeg`.
 * `--plot-contour/scatter/density` have been moved over to their own option group in `option_utils.py`, which `plot_posterior` and `plot_scatter` inherit from.
 * Some adjustments to font sizes.

Examples of different configurations:
 * [Scatter plot, no z-arg, no marginal plots](https://www.atlas.aei.uni-hannover.de/~cdcapano/LSC/projects/area_theorem/gw150914/inspiral-12ms_taylorf2_hmeco/posterior_plots_test/scatter_no_marginal.png)
 * [Scatter plot, points colored by SNR, no marginal plots](https://www.atlas.aei.uni-hannover.de/~cdcapano/LSC/projects/area_theorem/gw150914/inspiral-12ms_taylorf2_hmeco/posterior_plots_test/scatter_nomarginal_snr.png)
 * [Scatter plot, points colored by SNR, marginal plots](https://www.atlas.aei.uni-hannover.de/~cdcapano/LSC/projects/area_theorem/gw150914/inspiral-12ms_taylorf2_hmeco/posterior_plots_test/scatter_marginal-snr.png)
 * [Scatter plot with contours, points colored by SNR, marginal plots](https://www.atlas.aei.uni-hannover.de/~cdcapano/LSC/projects/area_theorem/gw150914/inspiral-12ms_taylorf2_hmeco/posterior_plots_test/scatter_marginal_contours-snr.png)
 * [Scatter plot with contours, no z-arg, marginal plots](https://www.atlas.aei.uni-hannover.de/~cdcapano/LSC/projects/area_theorem/gw150914/inspiral-12ms_taylorf2_hmeco/posterior_plots_test/scatter_marginal_contours.png)
 * [Density plot with contours, marginal plots](https://www.atlas.aei.uni-hannover.de/~cdcapano/LSC/projects/area_theorem/gw150914/inspiral-12ms_taylorf2_hmeco/posterior_plots_test/density_standard.png)
 * [2D density plot with contours, marginal plots](https://www.atlas.aei.uni-hannover.de/~cdcapano/LSC/projects/area_theorem/gw150914/inspiral-12ms_taylorf2_hmeco/posterior_plots_test/density_mp_ms.png)
 * [1D marginal plot](https://www.atlas.aei.uni-hannover.de/~cdcapano/LSC/projects/area_theorem/gw150914/inspiral-12ms_taylorf2_hmeco/posterior_plots_test/density_mchirp.png)

As this depends on PR #1050, I'm putting this on hold until that is merged.

cdcapano commented:
I think this is ready to merge now. @micamu I figure you're the best to review this since the changes are mostly to your code. Look good?

micamu commented:
Just two things (or one thing and one question):
1. In the scatter plots with no color, like [here](https://www.atlas.aei.uni-hannover.de/%7Ecdcapano/LSC/projects/area_theorem/gw150914/inspiral-12ms_taylorf2_hmeco/posterior_plots_test/scatter_marginal_contours.png), I would make the contours white or bright yellow or some other color that stands out more against the grey points.
2. What is the reason to name the movie plots after the sample number instead of the frame number?

micamu commented:
And if you really care so much about the beauty of the plots, I would change the grey points for dark blue, so that they match with the histograms :)

cdcapano commented:
Ok, I changed the default scatter color to navy with yellow contours if no z-arg is provided. Examples: [No contours, no marginals](https://www.atlas.aei.uni-hannover.de/~cdcapano/LSC/projects/area_theorem/gw150914/inspiral-12ms_taylorf2_hmeco/posterior_plots_test/scatter_no_marginal-update.png), [Contours and marginals](https://www.atlas.aei.uni-hannover.de/~cdcapano/LSC/projects/area_theorem/gw150914/inspiral-12ms_taylorf2_hmeco/posterior_plots_test/scatter_no_marginal-update_contour.png). Yellow was the best I could do with navy-colored points; it's difficult getting high contrast in this case. Look ok?

For the naming conventions output by `plot_movie`: I changed it to sample number so that you could easily combine the outputs of multiple different runs that used different start and end points (say, if we decide to parallelize it in condor). I added the zero-padding because it's difficult to get the frames sorted properly in bash when feeding them to ffmpeg without them; e.g., if you do a file glob with ls, the files end up sorting like `file1.png file10.png file11.png file2.png file 3.png`, etc. with the zeros it's `file01.png file02.png file03.png`, etc., as it should be.

cdcapano commented:
I wasn't happy with how difficult it is to see the contours with the navy/yellow combo, so I changed the defaults again. Here's the new default color schemes:

https://www.atlas.aei.uni-hannover.de/~cdcapano/LSC/projects/area_theorem/gw150914/inspiral-12ms_taylorf2_hmeco/posterior_plots_test/new_colors/

Happy @micamu?

micamu commented:
Yes, that looks good
